clippy_lints/src/trait_bounds.rs

```diff
@@ -167,6 +167,63 @@ impl<'tcx> LateLintPass<'tcx> for TraitBounds {
             }
         }
     }
+
+    fn check_ty(&mut self, cx: &LateContext<'tcx>, ty: &'tcx Ty<'tcx>) {
+        let TyKind::Ref(
+            ..,
+            MutTy {
+                ty: Ty {
+                    kind: TyKind::TraitObject(
+                        bounds,
+                        ..
+                    ),
+                    ..
+                },
+                ..
+            }
+        ) = ty.kind else { return; };
+
+        if bounds.len() < 2 {
+            return;
+        }
+
+        let mut bounds_span = bounds[0].span;
+
+        for bound in bounds.iter().skip(1) {
+            bounds_span = bounds_span.to(bound.span);
+        }
+
+        let mut seen_def_ids = FxHashSet::default();
```

Comment:
thought: it might be cheaper to just have two booleans for `Send` and `Sync` since those are the only two traits allowed in trait object bounds.

Reply:
Isn’t this approach more future proof?

Reply:
Also what about UnwindSafe/Unpin?

Reply:
Hm, that's fair then